model/countries/AU/AU-parsing-rules.yaml

```diff
@@ -0,0 +1,376 @@
+regex_definitions:
+  kBuildingValueRe:  # Regex for type "building"
+    regex_concat:
+      parts:
+      - regex_fragment: \d+  # House number
+      # Covering cases - house number with alphabetic suffix, house number range.
+      - regex_fragment: '(?:'
+      - regex_fragment: \s*[[:alpha:]]\b # Optional suffix for building ids with a letter suffix, e.g. 12a.
+      - regex_fragment: '|'
+      - regex_fragment: \s*?[-]\s*\d+ # Housnumber range (10-17)
+      - regex_fragment: ')?'
+      wrap_non_capture: false
+
+  kUnitTypeLiteralRe:  # Regex for "unit-type"
+    regex_fragment: (?:\b(?:apt\.?|apartment|se\.?|ste\.?|suite|unit|u\.?|room|rm\.?|shop|po box|ms\.?|mail shop|mb\.?|marine berth|vlla\.?|villa|we\.?|warehouse|office|off\.?|fy\.?|factory|sl\.?|stall|site|shed))
+
+  kUnitNameValueRe:  # Regex for "unit-name"
+    regex_fragment: (?:\d+\w?\b|\w\b)
+
+  kFloorLiteralRe:  # Regex for floor literal
+    regex_fragment: (?:\b(?:level|l\.?|floor|fl\.?|basement|b\.?|ground floor|g\.?|lower ground floor|lg\.?|upper ground floor|ug\.?|mezzanine|m\.?))
+
+capture_definitions:
+  StreetAddressDecomposition:
+    capture:
+      output: street-address-alternative-1
+      parts:
+      - capture_reference: ParseInBuildingLocation
+      - separator: {regex_fragment: '.*?'} 
+      - capture_reference: ParseBuildingLocation
+
+  ParseBuildingLocation:
+    capture:
+      output: building-location
+      parts:
+      - capture:
+          output: building
+          parts: [ {regex_reference: kBuildingValueRe} ]
+      - separator: {regex_reference: kWhitespaceSeparator}
+      - capture:
+          output: street
+          parts: [ {regex_reference: kMultipleLazyWordsRe} ]
+
+  ParseInBuildingLocation:
+    capture:
+      output: in-building-location
+      parts:
+        - capture_reference: ParseUnitWithMandatoryPrefix
+        - separator: {regex_reference: kWhitespaceSeparator}
+        - capture_reference: ParseFloorWithMandatoryPrefix
+      quantifier: MATCH_OPTIONAL
+
+  ParseUnitWithMandatoryPrefix:
+    capture:
+      output: unit
+      parts:
+      - capture:
+          output: unit-type
+          parts: [ {regex_reference: kUnitTypeLiteralRe} ]
+          quantifier: MATCH_OPTIONAL
+      - separator: {regex_reference: kWhitespaceSeparator}
+      - capture:
+          output: unit-name
+          parts: [ {regex_reference: kUnitNameValueRe} ]
+      quantifier: MATCH_OPTIONAL
+
+  ParseFloorWithMandatoryPrefix:
+    # match 'level \d+'
+    capture:
+      output: floor
+      prefix:
+        regex_concat:
+          parts:
+            - regex_reference: kFloorLiteralRe
+            - regex_fragment: '\s*'
+          wrap_non_capture: false
+      parts: [ {regex_fragment: '(?:\d+|.*?)'}]
```

Comment:
Isn't `.*?` a bit generous here? I would add `basement|b\.?|ground floor|g\.?|lower ground floor|lg\.?|upper ground floor|ug\.?|mezzanine|m\.?`

It would not be perfect because people would probably not write  "Floor basement" but just "basement". For that we would need to need to define a second copy `ParseFloorLiteralWithoutPrefix` and then change `ParseInBuildingLocation:` to try `ParseFloorWithMandatoryPrefix` and `ParseFloorLiteralWithoutPrefix` as alternatives.:

```
  ParseInBuildingLocation:
    capture:
      output: in-building-location
      parts:
        - capture_reference: ParseUnitWithMandatoryPrefix
        - separator: {regex_reference: kWhitespaceSeparator}
        - regex_fragment: '(?:'
        - capture_reference: ParseFloorWithMandatoryPrefix
        - regex_fragment: '|'
        - capture_reference: ParseFloorLiteralWithoutPrefix
        - regex_fragment: ')'
      quantifier: MATCH_OPTIONAL
```

I am not 100% sure it will work, we'll need to try.